--- gnulib/lib/glthread/lock.h  2019-01-16 11:48:44.000000000 +0200
+++ lock.h  2019-01-16 11:49:34.000000000 +0200
@@ -148,20 +148,7 @@
 #   pragma weak pthread_self
 #  endif

-#  if !PTHREAD_IN_USE_DETECTION_HARD
-    /* On most platforms, pthread_cancel or pthread_kill can be used to
-       determine whether libpthread is in use.
-       On newer versions of FreeBSD, however, this is no longer possible,
-       because pthread_cancel and pthread_kill got added to libc.  Therefore
-       use pthread_create to test whether libpthread is in use.  */
-#   if defined __FreeBSD__ || defined __DragonFly__ /* FreeBSD */
-#    pragma weak pthread_create
-#    define pthread_in_use() (pthread_create != NULL)
-#   else /* glibc, NetBSD, OpenBSD, IRIX, OSF/1, Solaris */
-#    pragma weak pthread_cancel
-#    define pthread_in_use() (pthread_cancel != NULL)
-#   endif
-#  endif
+# define pthread_in_use() 1

 # else
