diff --git a/ruby-augeas-0.5.0/ext/augeas/_augeas.c b/ruby-augeas-0.5.0/ext/augeas/_augeas.c
index 95d9e05024..7aac0eb96d 100644
--- a/ruby-augeas-0.5.0/ext/augeas/_augeas.c
+++ b/ruby-augeas-0.5.0/ext/augeas/_augeas.c
@@ -489,6 +489,7 @@ void Init__augeas() {
 
     /* Define the ruby class */
     c_augeas = rb_define_class("Augeas", rb_cObject) ;
+        rb_undef_alloc_func(c_augeas);
 
     /* Constants for enum aug_flags */
 #define DEF_AUG_FLAG(name) \
