--- selinuxswig_ruby_wrap.c.orig	2023-01-18 22:38:21.527547362 -0800
+++ selinuxswig_ruby_wrap.c	2023-01-18 22:38:26.887569066 -0800
@@ -1724,6 +1724,7 @@
 {
   /* register a new class */
   VALUE cl = rb_define_class("swig_runtime_data", rb_cObject);
+  rb_undef_alloc_func(cl);
   /* create and store the structure pointer to a global variable */
   swig_runtime_data_type_pointer = Data_Wrap_Struct(cl, 0, 0, pointer);
   rb_define_readonly_variable("$swig_runtime_data_type_pointer" SWIG_RUNTIME_VERSION SWIG_TYPE_TABLE_NAME, &swig_runtime_data_type_pointer);
