--- selinuxswig_ruby_wrap.c.orig	2024-06-11 20:02:30.112707265 +0000
+++ selinuxswig_ruby_wrap.c	2024-06-11 22:34:16.000151780 +0000
@@ -1510,7 +1510,7 @@
     _cSWIG_Pointer = rb_define_class_under(_mSWIG, "Pointer", rb_cObject);
     rb_undef_method(CLASS_OF(_cSWIG_Pointer), "new");
   }
-  rb_define_class_under(_mSWIG, klass_name, _cSWIG_Pointer);
+  rb_undef_alloc_func(rb_define_class_under(_mSWIG, klass_name, _cSWIG_Pointer));
   free((void *) klass_name);
 }
