diff --git a/configure.ac b/configure.ac
index e5454f117e..c809344f79 100644
--- a/configure.ac
+++ b/configure.ac
@@ -606,23 +606,6 @@ RUBY_WERROR_FLAG([
     cd .. && rm -fr tmp.$$.try_link
 ])
 
-: ${RPATHFLAG=''}
-rpathflag=''
-AS_IF([test x"${RPATHFLAG}" = x], [
-    AS_CASE(["$target_os"],
-	[aix*], [rpathflag='-blibpath:'],
-	[for rpathflag in -R "-rpath "; do
-	    AS_CASE("$rpathflag",
-		    [*" "], [AS_CASE(["${linker_flag}"],
-				     [*,], [rpathflag=`echo "$rpathflag" | tr ' ' ,`])])
-	    rpathflag="${linker_flag}${rpathflag}"
-	    RUBY_TRY_LDFLAGS([${rpathflag}.], [], [rpathflag=])
-	    AS_IF([test "x${rpathflag}" != x], [])
-        done])
-], [
-    rpathflag=`echo "$RPATHFLAG" | sed 's/%.*//'`
-])
-
 RUBY_TRY_LDFLAGS(-fdeclspec, [fdeclspec=yes], [fdeclspec=no])
 AS_IF([test "$fdeclspec" = yes], [
     RUBY_APPEND_OPTIONS(CFLAGS, -fdeclspec)
@@ -945,6 +928,7 @@ AC_ARG_WITH(opt-dir,
 		val=`echo "$PATH_SEPARATOR$withval" | sed "s|$PATH_SEPARATOR\([[^$PATH_SEPARATOR]*]\)| -I\1/include|g;s/^ //"`
 		CPPFLAGS="$CPPFLAGS $val"
 		val=`echo "$PATH_SEPARATOR$withval" | sed "s|$PATH_SEPARATOR\([[^$PATH_SEPARATOR]*]\)| -L\1/lib${rpathflag:+ $rpathflag\\\\1/lib}|g;s/^ //"`
+                val=`echo "$PATH_SEPARATOR$withval" | sed "s|$PATH_SEPARATOR\([[^$PATH_SEPARATOR]*]\)| -L\1/lib|g;s/^ //"`
 		LDFLAGS="$LDFLAGS $val"
 		LDFLAGS_OPTDIR="$val"
 		OPT_DIR="$withval"
@@ -2932,7 +2916,7 @@ AC_SUBST(LIBEXT)dnl
 AC_SUBST(ASMEXT, S)dnl
 
 STATIC=
-
+: ${PATHFLAG=''}
 : "dlopen" && {
   rb_cv_dlopen=unknown
   AC_MSG_CHECKING(whether OS depend dynamic link works)
@@ -3048,6 +3032,7 @@ STATIC=
                         TRY_LINK='$(CC) -oconftest $(INCFLAGS) -I$(hdrdir) $(CPPFLAGS)'
                         TRY_LINK="$TRY_LINK"' $(CFLAGS) $(src) $(LIBPATH) $(LDFLAGS) $(LOCAL_LIBS) $(LIBS)'
 			: ${LIBPATHENV=LIBPATH}
+                        RPATHFLAG=" ${linker_flag}-blibpath:%1\$-s:${prefix}/lib:${LIBPATH:-/usr/lib:/lib}"
 			: ${PRELOADENV=LDR_PRELOAD}
 			rb_cv_dlopen=yes],
 	[nto-qnx*], [	DLDFLAGS="$DLDFLAGS -L/lib -L/usr/lib -L/usr/local/lib"
@@ -3084,9 +3069,20 @@ STATIC=
   ])
 
   AS_IF([test "$enable_rpath:${RPATHFLAG}" = yes:], [
-      AS_IF([test "x$rpathflag" != x], [
-	  RPATHFLAG=" ${rpathflag}%1\$-s"
-      ])
+  if test "$enable_rpath" = yes; then
+    if test x"${RPATHFLAG}" = x; then
+      for rpathflag in -R "-rpath "; do
+       AS_CASE("$rpathflag",
+               [*" "], [AS_CASE(["${linker_flag}"],
+                                [*,], [rpathflag=`echo "$rpathflag" | tr ' ' ,`])])
+       rpathflag="${linker_flag}${rpathflag}"
+       RUBY_TRY_LDFLAGS([${rpathflag}.], [], [rpathflag=])
+       if test "x${rpathflag}" != x; then
+                 break
+       fi
+      done
+    fi
+  fi
   ])
 }
 
