diff --git a/lib/rubygems/compatibility.rb b/lib/rubygems/compatibility.rb
index f1d452e..4c5f422 100644
--- a/lib/rubygems/compatibility.rb
+++ b/lib/rubygems/compatibility.rb
@@ -13,7 +13,7 @@
 # TODO remove at RubyGems 4
 module Gem
   RubyGemsVersion = VERSION
-  deprecate_constant(:RubyGemsVersion)
+  deprecate_constant(:RubyGemsVersion) if respond_to?(:deprecate_constant)
 
   RbConfigPriorities = %w[
     MAJOR
@@ -30,7 +30,7 @@ module Gem
     ConfigMap = Hash.new do |cm, key|
       cm[key] = RbConfig::CONFIG[key.to_s]
     end
-    deprecate_constant(:ConfigMap)
+    deprecate_constant(:ConfigMap) if respond_to?(:deprecate_constant)
   else
     RbConfigPriorities.each do |key|
       ConfigMap[key.to_sym] = RbConfig::CONFIG[key]
