diff --git a/lib/sys/unix/sys/filesystem/functions.rb b/lib/sys/unix/sys/filesystem/functions.rb
index a821b15..a51b96f 100644
--- a/lib/sys/unix/sys/filesystem/functions.rb
+++ b/lib/sys/unix/sys/filesystem/functions.rb
@@ -7,7 +7,7 @@
 
       ffi_lib FFI::Library::LIBC
 
-      if RbConfig::CONFIG['host_os'] =~ /sunos|solaris|linux/i
+      if RbConfig::CONFIG['host_os'] =~ /sunos|solaris|x86_64.*linux/i
         attach_function(:statvfs, :statvfs64, [:string, :pointer], :int)
       else
         attach_function(:statvfs, [:string, :pointer], :int)
